TACC/Core-Portal
# Task/FP-1090: Put welcome message visited status in database, not localStorage

## Overview: ##

Put welcome message visited status in database, not localStorage

## Related Jira tickets: ##

* [FP-1090](https://jira.tacc.utexas.edu/browse/FP-1090)

## Summary of Changes: ##

- Updated IntroMessages saga, removing read/writes to localStorage and adding calls to newly created server side APIs  for storing and updating read IntroMessages.
- Added unit test for server side testing 

## Testing Steps: ##

1.  Update data model to create IntroMessages table.
    - `./mange.py migrate` in `core_portal_django` shell 
2. Start up development environment. Navigate to Dashboard
3. IntroMessages should appear appear at the top of the Dashboard, Data Files, Applications, Allocations, History, and UI pages.
4. Click the 'x' to dismiss each message.  Refresh the page.  The IntroMessage should not reappear. 

## UI Photos:
Dashboard with IntroMessage Visible
![Dashboard with IntroMessage Visible](https://user-images.githubusercontent.com/77903391/141156943-287de96f-52e2-404b-a6e6-20e16f686428.png)

Dashboard with IntroMessage Dismissed
![Dashboard with IntroMessage Dismissed](https://user-images.githubusercontent.com/77903391/141157233-de6b36ff-6f77-4de3-8574-3d950c4debe0.png)

## Notes: ##
Since the use/form of the IntroMessage is likely changing, I minimized the work on the front end, leaving the redux store unchanged from the way it worked with storing the IntroMessages' read/unread status in localStorage.  I accommodated for this on the server side by storing the IntroMessage as two columns: component (IntroMessage name) and unread (read/unread status).  unread == False means that a message has been read/dismissed by the user.  If a change to the way the redux store works, perhaps another Jira ticket could be created to tackle it?
